frontend/public/components/graphs/prometheus-graph.tsx

```diff
@@ -39,19 +38,22 @@ export const PrometheusGraphLink_: React.FC<PrometheusGraphLinkProps> = ({
   if (!query) {
     return <>{children}</>;
   }
-
+  const prometheusBaseURLPresent = !!window.SERVER_FLAGS.prometheusBaseURL;
+  const canAccessMonitoring = (canAccessNS && prometheusBaseURLPresent) || prometheusBaseURLPresent;
   const params = new URLSearchParams();
   params.set('query0', query);
 
   const url =
-    canAccessMonitoring && perspective === 'admin'
+    perspective === 'admin'
```

Comment:
Is this right? Can't normal users see monitoring for workloads in their own namespace even if they don't have the get namespace permission?

Reply:
I thought that if `canAccessNS === false` we cannot redirect the user to `/monitoring` otherwise they get a bunch of graphs with Forbidden errors.

Reply:
My understanding is that the admin query browser uses `prometheusBaseURL`, which will return errors for a normal user. Dev console uses `prometheusTenancyBaseURL`, which will work for a normal user. That's why we're linking to the developer view when the user doesn't have get namespace permission for the admin query borwser.

Regardless, I wouldn't expect to ever have a blank page. Is there a runtime error?

Reply:
It looks like we're reverting fix #3333 for https://bugzilla.redhat.com/show_bug.cgi?id=1770808

Reply:
Updated the PR. Now, wrapping the chart with ```PrometheusGraphLink``` only if ```prometheusBaseURL``` is present.

Reply:
As @spadgett mentioned, the dev console uses `prometheusTenancyBaseURL` rather than `prometheusBaseURL`.

So I think the logic should be like.
```
if (perspective === 'admin' && canAccessNS && window.SERVER_FLAGS.prometheusBaseURL) {
  url = `/monitoring/query-browser?${params.toString()}`;
} else if (window.SERVER_FLAGS.prometheusTenancyBaseURL) {
  url = `/dev-monitoring/ns/${namespace}/metrics?${params.toString()}`;
}
```

Reply:
Actually, if the `canAccess` check is being removed from `PageContents`, I think we can leave `PrometheusGraphLink` unchanged.

Reply:
I added check for ```prometheusTenancyBaseURL``` on ```Link``` so that if ```prometheusTenancyBaseURL``` is not present then we should not redirect the user to metrics page after clicking on the chart.

Reply:
@vikram-raj Adding the check isn't bad, but `prometheusBaseURL` and `prometheusTenancyBaseURL` will *always* be set in an OpenShift cluster. The only times those aren't set are when running against native k8s or running off-cluster, which is for development only. So it's unclear to me what bug that fixes.